PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/ClassifiedException.kt

```diff
@@ -0,0 +1,64 @@
+package com.microsoft.azure.hdinsight.common
+
+import com.microsoft.azure.hdinsight.common.logger.ILogger
+import com.microsoft.azure.hdinsight.spark.common.SparkJobException
+import org.apache.commons.lang.exception.ExceptionUtils
+import java.io.IOException
+
+const val ToolPackageSuffix: String = "com.microsoft.azure.hdinsight"
+const val EmptyLog: String = "Exception without detail message"
+
+enum class ExceptionType {
+    USER_EXCEPTION,
+    SYS_EXCEPTION,
+    TOOL_EXCEPTION
+}
+
+class ClassifiedException : Throwable, ILogger {
+    override val cause: Throwable?
+    override val message: String
+
+
+    constructor(msg: String) {
+        cause = null
+        message = msg
+    }
+
+    constructor(exp: Throwable) {
+        cause = exp.cause ?: exp
+        message = cause.message ?: EmptyLog
+    }
+
+
+    fun getStackTrace(): String {
+        if (cause == null) {
+            return message
+        }
+
+        return ExceptionUtils.getStackTrace(cause)
+    }
+
+    fun getExceptionType(): ExceptionType {
+        val expType =
+                when {
+                    cause is IOException -> ExceptionType.SYS_EXCEPTION
+                    cause is SparkJobException -> ExceptionType.USER_EXCEPTION
+                    stackTrace.contains(ToolPackageSuffix) -> ExceptionType.TOOL_EXCEPTION
+                    else -> ExceptionType.USER_EXCEPTION
+                }
+
+        return expType
+    }
+
+
+    fun logStackTrace() {
+        val stackTrace = stackTrace
+        val errMsg: String = if (!stackTrace.isEmpty()) {
```

Comment:
we can shorten the code to `val errMsg: String = (!stackTrace.isEmpty()) stackTrace else message`